### PR TITLE
#171, #172, #173, #177: Changed Priority to Optionality in all tables…

### PR DESCRIPTION
**Pull request by brynrhodes**
…. Added Type column to all tables to split type out from description. Fixed several references to Patient.id (changed to context.patientId) and corrected prefetch-template documentation based on new context tokens. Also changed User.id to userId for consistency with hook context names.

### CODE REVIEW

**Comment by isaacvetter** [comment]
Hey Bryn,

It might complicate merging a bit, but I think you should also update the hooks pages to address #177. 

Isaac

**Comment by isaacvetter** [comment]
Hey Bryn,

PR #181 is more targeted towards the context variable syntax change {{User.Id}} -> {{User}}.

Mind undoing those sets of changes from this PR?

Isaac

**Comment by isaacvetter** [comment]
Why is so much of the diff in hooks/index.md highlighted in red?

https://github.com/cds-hooks/docs/pull/183/files#diff-64ff89a786c99c7caf0aac385c90dfb2

**Comment by isaacvetter** [comment]
Is there an extra brace around the docs/hooks/medication-prescribe.md's example medications DSTU Bundle first MedicationOrder?

(This is probably my fault, not yours).

**Comment by kpshek** [comment]
> Why is so much of the diff in hooks/index.md highlighted in red?

@isaacvetter - There was a missing `}` in the preceding JSON example which was causing the parse issue. I've fixed that in c956457

**Comment by kpshek** [comment]
@isaacvetter - That JSON example was still a bit malformed. I fixed it and through it through a formatter (https://jsonformatter.curiousconcept.com/) in c7c8850

**Comment by brynrhodes** [comment]
@isaacvetter , I agree there's probably too much content to make the same page be a template, but I like the idea of a hook template. Should I add one to this PR or should that be a separate issue?

**Comment by isaacvetter** [comment]
Hey @brynrhodes ,

Undoubtedly a different issue, but perhaps one we should resolve quickly. I expect the push to create new hooks will boil over soon.

Isaac

File: docs/hooks/index.md [hunk 1]
```diff
@@ -82,52 +82,59 @@ Only the `encounterId` field in this example is eligible to be a prefetch token
...
 
 Describe the set of contextual data used by this hook. Only data logically and necessarily associated with the purpose of this hook should be represented in context.
 
-All fields defined by the hook's context MUST be defined in a table where each field is described by three attributes:
+All fields defined by the hook's context MUST be defined in a table where each field is described by the following attributes:
+
+- Field: The name of the field in the context JSON object.
+- Optionality: A string value of either `REQUIRED` or `OPTIONAL`
+- Prefetch Token: A string value of either `Yes` or `No`, indicating whether this field can be tokenized in a prefetch template.
+- Type: The type of the field in the context JSON object, expressed as the JSON type, or the name of a FHIR Resource type. Valid types are *string*, *number*, *object*, *array*, or the name of a FHIR resource type. When a field can be of multiple types, type names MUST be separated by a *pipe* (`|`)
```
Comment (kpshek) [inline docs/hooks/index.md]: What about types `true`, `false`, and `null` (all of which are allowed by JSON)?

Comment (brynrhodes) [reply docs/hooks/index.md]: Agreed, I added *boolean* to the list. Null is allowed for the values, but I don't think we'd want the type not to be specified, right?

Comment (kpshek) [reply docs/hooks/index.md]: Oh, good point! Yes, null shouldn't be a type 😄

**Pull request merged**